portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,33 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.*;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
+public final class DataServlet extends HttpServlet {
+
+  // Messages are hard-coded for testing purposes
+  private ArrayList<String> comments = new ArrayList<String>(
+      Arrays.asList("First comment", "Second comment", "Third comment")
+  );
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html");
-    response.getWriter().println("Hello Aradhya!");
+    String json = convertArrayListToJson(comments);
+    response.setContentType("application/json");
+    response.getWriter().println(json);
+  }
+
+  private String convertArrayListToJson(ArrayList<String> comments) {
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
```

Comment:
Nit: you could just return this instead of saving it in a variable because you don't ever need to reuse it

Reply:
Good point, I copied this from the tutorial where I assume they were being as verbose as possible. Will make that change.